Zend/zend_max_execution_timer.c

```diff
@@ -98,6 +109,16 @@ void zend_max_execution_timer_shutdown(void) /* {{{ */
 	if (err != 0) {
 		zend_strerror_noreturn(E_ERROR, errno, "Could not delete timer");
 	}
+
+	struct sigaction ign_sigrtmin;
+	ign_sigrtmin.sa_handler = SIG_IGN;
```

Comment:
Not relevant here, but `SIG_DFL` might be preferable.

Reply:
The only way to clear these is with `SIG_IGN`. The handler should already be set to `SIG_DFL` long before shutdown is called in FrankenPHP. I'll go ahead and answer your other question here because it is relevant:

>  I'm not sure I understand why the signals are never delivered. Can you clarify?

To get a hint at what is going on, we have to check the man page for [timer_delete(2)](https://www.man7.org/linux/man-pages/man2/timer_delete.2.html), and pay close attention to this little nugget:

> The treatment of any pending signal generated by the deleted timer is unspecified.

In practice (at least with all the kernels I have tested with), the signals just chill out in the SigQ forever until SigQ becomes full and starts giving an error. The only way to clear them is to set the signal handler to `SIG_IGN`.

I would have to venture into the kernel source code to actually deeply understand how this happens, but this is the observed behavior.

Reply:
Thank you for the explanation. A comment about this behavior would definitely be helpful.